apps/metadata-editor/src/app/edit/components/publish-button/publish-button.component.html

```diff
@@ -11,12 +13,49 @@
     <mat-spinner [diameter]="16"></mat-spinner>
   </gn-ui-button>
   <gn-ui-button
+    cdkOverlayOrigin
+    #actionMenuButton
     *ngSwitchCase="'upToDate'"
     type="secondary"
     [disabled]="true"
-    (buttonClick)="saveRecord()"
+    (buttonClick)="verifyPublishConditions()"
     [matTooltip]="'editor.record.upToDate' | translate"
   >
     <ng-icon name="matCheckCircleOutline"></ng-icon>
   </gn-ui-button>
+  <ng-template #template>
+    <div
+      data-test="publish-warning"
+      class="w-96 p-5 flex flex-col gap-3 mt-2 border border-gray-100 bg-white shadow-2xl rounded-2xl mr-2"
+    >
+      <span
+        class="text-center"
+        translate
+        [translateParams]="{
+          date: formatDate(publishWarning.date),
```

Comment:
I think you could use angular's date pipe.

Reply:
The issue with Angular's date pipe is that we cannot format it as "date at time"/ "jour à heure" (with the joining term) in the current langugage. Also, the test was not really useful since I was just using `toLocaleString`, which does not really need testing anyway.